daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+        fields_to_be_downloaded = [field for field in required_fields if not all([os.path.exists(stor_dir + '{o}'.format(o=o)) for o in obs_dict[field]])]
+        
+        if not self._download_done:
+            # If only one core is to be used, then it's simply a case of a nested loop through ObsIDs and instruments
+            if num_cores == 1:
+                with tqdm(total=len(fields_to_be_downloaded), desc="Downloading {} data".format(self._pretty_miss_name)) as download_prog:
+                    for field in fields_to_be_downloaded:
+                        # Use the internal static method I set up which both downloads and unpacks the eROSITACalPV data
+                        self._download_call(self, field=field)
+                        # Update the progress bar
+                        download_prog.update(1)
+
+            elif num_cores > 1:
+                # List to store any errors raised during download tasks
+                raised_errors = []
+
+                # This time, as we want to use multiple cores, I also set up a Pool to add download tasks too
+                with tqdm(total=len(fields_to_be_downloaded ), desc="Downloading {} data".format(self._pretty_miss_name)) \
+                        as download_prog, Pool(num_cores) as pool:
+
+                    # The callback function is what is called on the successful completion of a _download_call
+                    def callback(download_conf: Any):
+                        """
+                        Callback function for the apply_async pool method, gets called when a download task finishes
+                        without error.
+
+                        :param Any download_conf: The Null value confirming the operation is over.
+                        """
+                        nonlocal download_prog  # The progress bar will need updating
+                        download_prog.update(1)
+
+                    # The error callback function is what happens when an exception is thrown during a _download_call
+                    def err_callback(err):
+                        """
+                        The callback function for errors that occur inside a download task running in the pool.
+
+                        :param err: An error that occurred inside a task.
+                        """
+                        nonlocal raised_errors
+                        nonlocal download_prog
+
+                        if err is not None:
+                            # Rather than throwing an error straight away I append them all to a list for later.
+                            raised_errors.append(err)
+                        download_prog.update(1)
+
+                    # Again nested for loop through ObsIDs and instruments
+                    for field in fields_to_be_downloaded:
+                        # Add each download task to the pool
+                        pool.apply_async(self._download_call,
```

Comment:
Yeah so this is what causes the issue, as you don't pass self - again won't be necessary once download_call is static

Reply:
yeah, all these comments I think are fully resolved, I can run the download() function after I made it a proper static method